internal/helmupdater/helmupdater_test.go

```diff
@@ -0,0 +1,101 @@
+package helmupdater_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/rancher/fleet/internal/helmupdater"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type fsNodeSimple struct {
+	name     string
+	children []fsNodeSimple
+	isDir    bool
+}
+
+// createDirStruct generates and populates a directory structure which root is node, placing it at basePath.
+func createDirStruct(t *testing.T, basePath string, node fsNodeSimple) {
+	path := filepath.Join(basePath, node.name)
+
+	if !node.isDir {
+		_, err := os.Create(path)
+		require.NoError(t, err)
+		return
+	}
+
+	err := os.Mkdir(path, 0777)
+	require.NoError(t, err)
+
+	for _, c := range node.children {
+		createDirStruct(t, path, c)
+	}
+}
+
+func TestFoundChartYamlInDirectory(t *testing.T) {
+	cases := []struct {
+		name               string
+		testDir            string
+		directoryStructure fsNodeSimple
+		expectedResult     bool
+	}{
+		{
+			name:    "simple case having Chart yaml file in the root",
+			testDir: "simpleok",
+			directoryStructure: fsNodeSimple{
+				name: helmupdater.ChartYaml,
```

Comment:
My intention was to make it clear that those tests were using the value defined in the package: `helmupdater.ChartYaml` i.e. the expected one. 
If someone changes the value expected in the main package I would expect the test to pass without having to go and change literals explicitly... sometimes that is a tedious task and in this case integration tests would fail anyway if the change was done by mistake. 

BUT it could be seen the other way around, as noted here. :) 
I changed it as I don't think that value is going to change in the future.